src/core/init/duration_updater.ts

```diff
@@ -58,9 +65,22 @@ export default function DurationUpdater(
       if (isNaN(mediaSource.duration) || !isFinite(mediaSource.duration) ||
           Math.abs(mediaSource.duration - newDuration) > 0.01)
       {
+        if (mediaSource.readyState === "closed") {
+          log.warn("Init: can't update duration on closed MediaSource. " +
+                   "Waiting for MediaSource to open.");
+          return new Observable((obs) => {
+            mediaSource.onsourceopen = () => {
+              log.info("Init: Updating duration", newDuration);
+              mediaSource.duration = newDuration;
+              obs.next();
+              obs.complete();
+            };
+          });
```

Comment:
Shouldn't we do nothing if that Observable is unsubcribed from ?
If so why not just using `fromEvent` like for other event-based observables (also using addEventListener, as fromEvent is doing behind the scene) ensure that we're not overwritting anything